### DISCUSSION TITLE
Try not to loose the already performed indexation work after Major file and folder reorganization

### DISCUSSION THREAD

**Discussion by rezzalex** [photoprism/photoprism#2319] [Q&A]
Hello,

First, Thank you for this wonderful Application !

I have approx 65K files in my photo folder, with 2 Google Take-Out and beside that, A LOT of duplicates, for approx 217Go
I have tested [Phokup](https://github.com/ivandokov/phockup), in "copy" mode, to evaluate my duplicates. I could have in the end "only" 57K real unique photos.

The problem is that I have started to Index my initial and not optimized folder, and it has run all day long for approx 5 days. I am now at almost 26K photos indexed.

I am afraid to lose all this indexation work if I simply replace my current not optimized photo folder with the new optimized one, in PhotoPrism. I am afraid this would be a new indexation for Photoprism and that It takes 3 new weeks to reindex everything...

My Goals for optimizing my photo Folder will be : 

1.  no duplicate (most likely, keep the biggest file)
2. Keep all already available Tags
3. Add the Tag "photoTakenTime" if not present (most likely based on "LastModifyDate" )
4. Keep the Album folder structure of Google Photo takeouts, both in real folders and potentially with the script https://github.com/inthreedee/photoprism-transfer-album
5. Try to keep additional text informations in folder's name when present

Thank you for any advice or help

**Comment by graciousgrey** [comment]
In general, if you index your originals folder and afterwards delete files from it or move files, PhotoPrism detects those changes during the next indexing.
Avoid making those changes while the indexer is running!

We have no experience with Phockup so we can't tell you if it is a good idea. 
- It might delete files that are  associated with sidecar files
- It might delete files that are part of the google takeout albums you want to re-create with the community script

**Reply by rezzalex** [reply]
Hello.

Maybe, to be more precise, I can sum up what phockup is doing.
- Renaming files by date
- Reorganizing folders by date
- Deleting duplicates

It can be done in the same initial folder or by copying selected and renamed files elsewhere.

So what will happen in Photoprism if a file is renamed and changed of folder ?

What if this file had duplicates, and Photoprism previously selected one among them as unique photo,  and this one is deleted but another duplicate file remain ?

Thank you

**Reply by lastzero** [reply]
In general, PhotoPrism will pick up moved / renamed files. However, if ALL file names change in such a way then we honestly can not tell you exactly what will happen. It seems best to perform such major changes before you start indexing.

**Reply by rezzalex** [reply]
Thank you @lastzero and @graciousgrey for your answers.

I think I will try to remove my duplicates "manually", and not change my existing folder and filename structure, while the indexation continues (31 300 files until now). And ensuring I have the tag "photo taken time" everywhere.

When the 2 will be over, I will search for restructuring the filename and folder
